src/response/response.rs

```diff
@@ -51,9 +51,14 @@ impl Response {
             .headers(Some(CONTENT_LENGTH).zip(len))
     }
 
+    /// Build a response from a stream of `Result<Frame<Bytes>, Error>`.
+    ///
+    /// If you want to use a stream that is `!Unpin`, you can use the [`Boxed`]
+    /// body in combination with a [`ResponseBuilder`].
+    ///
     pub fn stream<S, E>(stream: S) -> ResponseBuilder
```

Comment:
This is an optimized path for the common case where a `Stream` is `Unpin`. We don't take any functionality away from users by imposing this requirement. You can still use an `!Unpin` stream with a `Boxed` body.

```rust
let stream = NotUnpinStream::new();
let body = Boxed::new(Box::pin(stream));

Response::build().body(body).finish()
```